client-side/src/app/components/addon/addon.component.ts

```
@@ -123,7 +130,7 @@ export class AddonComponent implements OnInit {
                     title: this.translate.instant("Delete"),
                     handler: async (objs) => {
                         objs = objs.map(obj => ({Key: obj.Key, Hidden: true}));
-                        this.todoservice.upsertTodos(objs)
+                        await this.todoservice.upsertTodos(objs)
```

If you are `await`ing this call you do not need to put the `reload` in the finally block.
Also don't use finally b/c it will discard errors thrown from the promise

---

Done